### ISSUE TITLE
Typescript typings for package

### ISSUE THREAD

**Issue by milad2golnia** [lynxtaa/awesome-graphql-client#81]
**Is your feature request related to a problem? Please describe.**
When using this package in a typescript project I receive following error:
```
/path-to-project/node_modules/awesome-graphql-client/dist/index.modern.mjs' implicitly has an 'any' type.
  There are types at '/path-to-project/node_modules/awesome-graphql-client/dist/index.d.ts', but this result could not be resolved when respecting package.json "exports". The 'awesome-graphql-client' library may need to update its package.json or typings.
```
NOTE: This problem is produced by vscode.

**Describe the solution you'd like**
I prefer to have a complete typing when using project or there should be a package like `@types/awesome-graphql-client` to add necessary typings.

**Comment by lynxtaa** [comment]
This project is written in Typescript, so we provide all the typings. And for me all the typings work with VSCode. Could you please provide more details?

**Comment by lynxtaa** [comment]
Closing since no reply

**Comment by onionhammer** [comment]
I'm also seeing this, perhaps the bundling isnt quite right @lynxtaa

**Comment by lynxtaa** [comment]
@onionhammer Could you please provide more details? Maybe small reproduction repository? I've tested it my projects and see no issues

**Comment by lynxtaa** [comment]
@onionhammer I think it depends on `tsconfig.json`, please share yours

**Comment by onionhammer** [comment]
@lynxtaa stock from npm create vite@latest test-build-project -- --template react-ts provides this:

```json
{
  "compilerOptions": {
    "target": "ES2020",
    "useDefineForClassFields": true,
    "lib": ["ES2020", "DOM", "DOM.Iterable"],
    "module": "ESNext",
    "skipLibCheck": true,

    /* Bundler mode */
    "moduleResolution": "bundler",
    "allowImportingTsExtensions": true,
    "resolveJsonModule": true,
    "isolatedModules": true,
    "noEmit": true,
    "jsx": "react-jsx",

    /* Linting */
    "strict": true,
    "noUnusedLocals": true,
    "noUnusedParameters": true,
    "noFallthroughCasesInSwitch": true
  },
  "include": ["src"],
  "references": [{ "path": "./tsconfig.node.json" }]
}

```

**Comment by lynxtaa** [comment]
Thanks! I've found the culprit, will release the fixed version today :)

**Comment by lynxtaa** [comment]
I've published awesome-graphql-client@1.0.0 , decided to switch the bundler for tsup. Hope it helps :)

**Comment by onionhammer** [comment]
@lynxtaa so far so good, thanks for the quick turnaround; I had no idea what a can of worms file uploading was :S
